src/GitHubExtension/DeveloperId/CredentialVault.cs

```diff
@@ -97,26 +91,29 @@ internal static PasswordCredential GetCredentialFromLocker(string loginId)
             else
             {
                 Log.Logger()?.ReportInfo("No credentials found for this DeveloperId");
-                throw new ArgumentOutOfRangeException(loginId);
+                return null;
             }
 
             var accessTokenInChars = new char[credentialObject.CredentialBlobSize / 2];
             Marshal.Copy(credentialObject.CredentialBlob, accessTokenInChars, 0, accessTokenInChars.Length);
 
-            var accessToken = new SecureString();
+            // convert accessTokenInChars to string
+            string accessTokenString = new (accessTokenInChars);
+
             for (var i = 0; i < accessTokenInChars.Length; i++)
             {
-                accessToken.AppendChar(accessTokenInChars[i]);
-
                 // Zero out characters after they are copied over from an unmanaged to managed type.
                 accessTokenInChars[i] = '\0';
             }
 
-            accessToken.MakeReadOnly();
-
-            var credential = new PasswordCredential(CredentialVaultConfiguration.CredResourceName, loginId, new NetworkCredential(string.Empty, accessToken).Password);
+            var credential = new PasswordCredential(credentialResourceName, loginId, accessTokenString);
             return credential;
         }
+        catch (Exception)
+        {
+            Log.Logger()?.ReportInfo($"Retrieving credentials from Credential Manager has failed unexpectedly");
+            throw new Win32Exception(Marshal.GetLastWin32Error());

```

Comment:
Should this have a result object?  If this isn't caught by DevHome, it'll crash.

Reply:
Good catch. Most callers of this function handle exceptions and return appropriate result objects. I did catch one place where this isn't caught correctly. Will fix.